spec/data/knife_subcommand/test_yourself.rb

```diff
@@ -11,6 +11,8 @@ class << self
 
     option :scro, :short => '-s SCRO', :long => '--scro SCRO', :description => 'a configurable setting'
 
+    option :with_proc, :long => '--with-proc VALUE', proc: Proc.new { |v| Chef::Config[:knife][:with_proc] = v }
```

Comment:
I would prefer to remove this as a way of writing cli options.  AFIAK this was the way that people tried to "make the cli win" circa chef-0.10.x and then we fixed most of that, but never went through and scrubbed all the Procs out, but if we're going to do anything I'd prefer to remove every single one of these Procs from the codebase.